test/python/primitives/test_backend_estimator_dev.py

```diff
@@ -0,0 +1,924 @@
+from qiskit.transpiler.passes import ApplyLayout, SetLayout
+from qiskit.test import QiskitTestCase
+
+BACKENDS = [FakeNairobi(), FakeNairobiV2()]
+
+
+################################################################################
+## AUXILIARY
+################################################################################
+def measurement_circuit_examples() -> Iterator[tuple[list[str], QuantumCircuit]]:
+    """Generator of commuting Paulis and corresponding measurement circuits.
+
+    Yields:
+        - List of commuting Pauli strings
+        - Quantum circuit to measure said Paulis
+    """
+    I = QuantumCircuit(1, 1)  # pylint: disable=invalid-name
+    I.measure(0, 0)
+    yield ["I", "Z"], I
+
+    X = QuantumCircuit(1, 1)  # pylint: disable=invalid-name
+    X.h(0)
+    X.measure(0, 0)
+    yield ["X", "I"], X
+
+    Y = QuantumCircuit(1, 1)  # pylint: disable=invalid-name
+    Y.sdg(0)
+    Y.h(0)
+    Y.measure(0, 0)
+    yield ["Y", "I"], Y
+
+    Z = QuantumCircuit(1, 1)  # pylint: disable=invalid-name
+    Z.measure(0, 0)
+    yield ["Z", "I"], Z
+
+    II = QuantumCircuit(2, 1)  # pylint: disable=invalid-name
+    II.measure(0, 0)
+    yield ["II"], II
+
+    IY = QuantumCircuit(2, 1)  # pylint: disable=invalid-name
+    IY.sdg(0)
+    IY.h(0)
+    IY.measure(0, 0)
+    yield ["IY", "II"], IY
+
+    XY = QuantumCircuit(2, 2)  # pylint: disable=invalid-name
+    XY.h(1)
+    XY.sdg(0)
+    XY.h(0)
+    XY.measure([0, 1], [0, 1])
+    yield ["XY", "II", "XI", "IY"], XY
+
+    XX = QuantumCircuit(2, 2)  # pylint: disable=invalid-name
+    XX.h(1)
+    XX.h(0)
+    XX.measure([0, 1], [0, 1])
+    yield ["XX", "IX", "XI", "II"], XX
+
+    ZZ = QuantumCircuit(2, 2)  # pylint: disable=invalid-name
+    ZZ.measure([0, 1], [0, 1])
+    yield ["ZZ", "IZ", "ZI", "II"], ZZ
+
+    XYZ = QuantumCircuit(3, 3)  # pylint: disable=invalid-name
+    XYZ.h(2)
+    XYZ.sdg(1)
+    XYZ.h(1)
+    XYZ.measure([0, 1, 2], [0, 1, 2])
+    yield ["XYZ", "XII", "IYI", "IIZ", "XIZ", "III"], XYZ
+
+    YIX = QuantumCircuit(3, 2)  # pylint: disable=invalid-name
+    YIX.sdg(2)
+    YIX.h(2)
+    YIX.h(0)
+    YIX.measure([0, 2], [0, 1])
+    yield ["YIX", "IIX", "YII", "III"], YIX
+
+    IXII = QuantumCircuit(4, 1)  # pylint: disable=invalid-name
+    IXII.h(2)
+    IXII.measure(2, 0)
+    yield ["IXII", "IIII"], IXII
+
+
+def circuit_composition_examples() -> Iterator[
+    tuple[QuantumCircuit, QuantumCircuit, QuantumCircuit]
+]:
+    """Generator of base and measurement circuits, and respective composition.
+
+    Yields:
+        - Transpiled base circuit: with a `final_layout` entry in its metadata
+        - Measurement circuit: before transpilation (i.e. no layout applied)
+        - Transpiled measurement circuit: with `final_layout` applied
+    """
+    target_qubits = 2
+    layout_intlist = (0,)
+    measured_qubits = (0,)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
```

Comment:
This is very minor, but why not just have all these lines be `build_composition_data(2, (0,), (0,))`?  You'd be able to read the whole function in one screen if we did that and easily count the output cases, rather than having to scroll through several to spot it on a failing test.

Reply:
After revisiting this I agree with you that this is no longer necessary so I removed the function and place the function calls directly in `@data`